python-package/xgboost/spark/core.py

```diff
@@ -349,9 +350,7 @@ def _validate_params(self) -> None:
             )
 
         tree_method = self.getOrDefault(self.getParam("tree_method"))
-        if (
-            self.getOrDefault(self.use_gpu) or use_cuda(self.getOrDefault(self.device))
-        ) and not _can_use_qdm(tree_method):
+        if tree_method == "exact":
```

Comment:
according to the raise msg, shouldn't add "use_gpu and tree_method=="exact"?

Reply:
Good catch, changed to `not supported on distributed systems`.